addons/calendar/views/calendar_views.xml

```diff
@@ -39,24 +39,20 @@
             <form string="Calendar Alarm">
                 <sheet>
                     <group>
-                        <group>
+                        <group name="left_details">
                             <field name="name" invisible="1"/>
                             <field name="alarm_type"/>
+                            <field name="mail_template_id" attrs="{'invisible': [('alarm_type','!=','email')], 'required': [('alarm_type', '=', 'email')]}"
```

Comment:
Missing some spaces in domain parts (see it is a copié-collé but quand même, let us add them)